packages/date-time/src/components/Calendar/CalendarYear/CalendarYear.base.tsx

```diff
@@ -0,0 +1,458 @@
+    });
+
+    return (
+      <div className={css('ms-DatePicker-decadeComponents', classNames.yearNavigationButtonsContainer)}>
+        <CalendarYearNavPrev {...this.props} />
+        <CalendarYearNavNext {...this.props} />
+      </div>
+    );
+  }
+}
+class CalendarYearTitle extends React.Component<ICalendarYearHeaderProps, {}> {
+  public render(): JSX.Element {
+    const { styles, theme, className, fromYear, toYear, onHeaderSelect, strings } = this.props;
+
+    const classNames = getClassNames(styles, {
+      theme: theme!,
+      className: className,
+      hasHeaderClickCallback: !!this.props.onHeaderSelect
+    });
+
+    if (onHeaderSelect) {
+      const rangeAriaLabel = (strings || DefaultCalendarYearStrings).rangeAriaLabel;
+      const ariaLabel = rangeAriaLabel
+        ? typeof rangeAriaLabel === 'string'
+          ? (rangeAriaLabel as string)
+          : (rangeAriaLabel as ICalendarYearRangeToString)(this.props)
+        : undefined;
+      return (
+        <button
+          className={css('ms-DatePicker-currentDecade js-showYearPicker', classNames.currentYearButton)}
+          onClick={this._onHeaderSelect}
+          onKeyDown={this._onHeaderKeyDown}
+          aria-label={ariaLabel}
+          role="button"
+          tabIndex={0}
+        >
+          {this._onRenderYear(fromYear)} - {this._onRenderYear(toYear)}
+        </button>
+      );
+    }
+    return (
+      <div className={classNames.currentYear}>
+        {this._onRenderYear(fromYear)} - {this._onRenderYear(toYear)}
+      </div>
+    );
+  }
+  private _onHeaderSelect = () => {
+    if (this.props.onHeaderSelect) {
+      this.props.onHeaderSelect();
+    }
+  };
+  private _onHeaderKeyDown = (ev: React.KeyboardEvent<HTMLElement>) => {
+    if (this.props.onHeaderSelect && (ev.which === KeyCodes.enter || ev.which === KeyCodes.space)) {
+      this.props.onHeaderSelect();
+    }
+  };
+  private _onRenderYear = (year: number) => {
+    if (this.props.onRenderYear) {
+      return this.props.onRenderYear(year);
+    }
+    return year;
+  };
+}
+class CalendarYearHeader extends BaseComponent<ICalendarYearHeaderProps, {}> {
+  public render(): JSX.Element {
+    const { styles, theme, className } = this.props;
+
+    const classNames = getClassNames(styles, {
+      theme: theme!,
+      className: className,
+      hasHeaderClickCallback: !!this.props.onHeaderSelect
+    });
+
+    return (
+      <div className={css('ms-DatePicker-header', classNames.headerContainer)}>
+        {this._onRenderTitle()}
+        {this._onRenderNav()}
+      </div>
+    );
+  }
+  private _onRenderTitle = () => {
+    if (this.props.onRenderTitle) {
+      return this.props.onRenderTitle(this.props);
+    }
+    return <CalendarYearTitle {...this.props} />;
+  };
+  private _onRenderNav = () => {
+    return <CalendarYearNav {...this.props} />;
+  };
+}
+export interface ICalendarYearState {
+  fromYear: number;
+  navigatedYear?: number;
+  selectedYear?: number;
+}
```

Comment:
Here is an exported interface in the middle of the file.. is there a reason it's here?

Reply:
Direct copy from other component. Will move to top.